lib/randomua/src/main/java/eu/kanade/tachiyomi/lib/randomua/RandomUserAgent.kt

```diff
@@ -0,0 +1,119 @@
+package eu.kanade.tachiyomi.lib.randomua
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.NetworkHelper
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.Response
+import uy.kohesive.injekt.injectLazy
+import java.io.IOException
+
+private class RandomUserAgent(
+    private val userAgentType: UserAgentType,
+    private val customUA: String?,
+    private val filterInclude: List<String>,
+    private val filterExclude: List<String>,
+) : Interceptor {
+
+    private var userAgent: String? = null
+
+    private val json: Json by injectLazy()
+
+    private val network: NetworkHelper by injectLazy()
+
+    private val client = network.client
+
+    override fun intercept(chain: Interceptor.Chain): Response {
+        try {
+            val originalRequest = chain.request()
+
+            val newUserAgent = getUserAgent()
+                ?: return chain.proceed(originalRequest)
+
+            val originalHeaders = originalRequest.headers
+
+            val modifiedHeaders = originalHeaders.newBuilder()
+                .set("User-Agent", newUserAgent)
+                .build()
+
+            return chain.proceed(
+                originalRequest.newBuilder()
+                    .headers(modifiedHeaders)
+                    .build()
+            )
+        } catch (e: Exception) {
+            throw IOException(e.message)
+        }
+    }
+
+    private fun getUserAgent(): String? {
+        if (userAgentType == UserAgentType.OFF) {
+            return customUA?.ifBlank { null }
+        }
+
+        if (!userAgent.isNullOrEmpty()) return userAgent
+
+        val uaResponse = client.newCall(GET(UA_DB_URL)).execute()
+
+        if (!uaResponse.isSuccessful) {
+            uaResponse.close()
+            return null
+        }
+
+        val userAgentList = uaResponse.use { json.decodeFromString<UserAgentList>(it.body.string()) }
+
+        return when (userAgentType) {
+            UserAgentType.DESKTOP -> userAgentList.desktop
+            UserAgentType.MOBILE -> userAgentList.mobile
+            else -> error("Expected UserAgentType.Desktop or UserAgentType.Mobile got UserAgentType.${userAgentType.name}")
+        }
+            .filter {
+                filterInclude.isEmpty() || filterInclude.any { filter ->
+                    it.contains(filter, ignoreCase = true)
+                }
+            }
+            .filterNot {
+                filterExclude.isNotEmpty() && filterExclude.any { filter ->
+                    it.contains(filter, ignoreCase = true)
+                }
+            }
```

Comment:
```suggestion
            .filterNot {
                filterExclude.any { filter ->
                    it.contains(filter, ignoreCase = true)
                }
            }
```
Safe to not check here